plugins/inputs/spark/spark.go

```diff
@@ -0,0 +1,441 @@
+				fields[value] = values[value]
+			}
+		}
+	} else {
+		for k, v := range values {
+			if v != nil {
+				fields[k] = v
+			}
+		}
+	}
+	c.acc.AddFields(class, fields, tags)
+
+}
+
+func (j javaMetric) addTagsFields(out map[string]interface{}) {
+	fmt.Println(out["request"])
+	request := out["request"].(map[string]interface{})
+	var mbean = request["mbean"].(string)
+	var mbeansplit = strings.Split(mbean, "=")
+	var class = mbeansplit[1]
+
+	if valuesMap, ok := out["value"]; ok {
+		if class == "Memory" {
+			addJavaMetric("spark_HeapMemoryUsage", j, valuesMap.(map[string]interface{}))
+		} else if class == "Threading" {
+			addJavaMetric("spark_Threading", j, valuesMap.(map[string]interface{}))
+		} else {
+			fmt.Printf("Missing key in '%s' output response\n%v\n",
+				j.metric, out)
+			return
+		}
+
+	}
+}
+
+func addSparkMetric(mbean string, c sparkMetric,
+	values map[string]interface{}) {
+
+	tags := make(map[string]string)
+	fields := make(map[string]interface{})
+
+	tokens := parseJmxMetricRequest(mbean)
+	addTokensToTags(tokens, tags)
+	tags["spark_host"] = c.host
+
+	addValuesAsFields(values, fields, tags["mname"])
+	c.acc.AddFields(tokens["class"]+tokens["type"], fields, tags)
+
+}
+
+func (c sparkMetric) addTagsFields(out map[string]interface{}) {
+	if valuesMap, ok := out["value"]; ok {
+		for k, v := range valuesMap.(map[string]interface{}) {
+			addSparkMetric(k, c, v.(map[string]interface{}))
+		}
+	} else {
+		fmt.Printf("Missing key 'value' in '%s' output response\n%v\n",
+			c.metric, out)
+		return
+	}
+
+}
+
+func addYarnMetric(c yarnMetric, value map[string]interface{}, metrictype string) {
+
+	tags := make(map[string]string)
+	fields := make(map[string]interface{})
+	tags["yarn_host"] = c.host
+	for key, val := range value {
+		fields[key] = val
+	}
+	c.acc.AddFields(metrictype, fields, tags)
+}
+
+func (c yarnMetric) addTagsFields(out map[string]interface{}) {
+
+	if valuesMap, ok := out["clusterMetrics"]; ok {
+		addYarnMetric(c, valuesMap.(map[string]interface{}), "spark_clusterMetrics")
+	} else if valuesMap, ok := out["clusterInfo"]; ok {
+		addYarnMetric(c, valuesMap.(map[string]interface{}), "spark_clusterInfo")
+	} else if valuesMap, ok := out["apps"]; ok {
+		for _, value := range valuesMap.(map[string]interface{}) {
+			for _, vv := range value.([]interface{}) {
+				addYarnMetric(c, vv.(map[string]interface{}), "spark_apps")
+			}
+		}
+	} else if valuesMap, ok := out["nodes"]; ok {
+		for _, value := range valuesMap.(map[string]interface{}) {
+			for _, vv := range value.([]interface{}) {
+				addYarnMetric(c, vv.(map[string]interface{}), "spark_nodes")
+			}
+		}
+	} else {
+		fmt.Printf("Missing the required key in output response\n%v\n", out)
+		return
+	}
+
+}
+
+func (j *Spark) SampleConfig() string {
+	return `
+  ## Spark server exposing jolokia read service
+  SparkServer = ["127.0.0.1:8778"] #optional
+  ## Server running Yarn Resource Manager
+  YarnServer = "127.0.0.1:8088" #optional
```

Comment:
Yarn is a central resource manager for the spark cluster. While spark cluster may have many nodes.